docs/madr/decisions/004-metric-filter.md

```diff
@@ -0,0 +1,136 @@
+# Metrics filtering
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/1755
+
+## Context and Problem Statement
+
+Envoy by default produces a lot of metrics. You may be interested only in subset of them to not stress your metric infrastructure.
+We need to have a mechanism to filter metrics.
+
+## Considered Options
+
+* Filter metrics dynamically on scrape request
+* Filter metrics statically on Envoy bootstrap
+* Customizable bootstrap
+
+## Decision Outcome
+
+Chosen option:
+"Filter metrics dynamically on scrape request", because it follows Kuma philosophy of being as dynamic as possible.
+Additionally, as a followup we should implement "Customizable bootstrap", because it both enables static metrics and other advanced use cases. 
+
+## Pros and Cons of the Options
+
+### Filter metrics dynamically on scrape request
+
+UX:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+  name: default
+spec:
+  metrics:
+    enabledBackend: prometheus-1
+    backends:
+    - name: prometheus-1
+      type: prometheus
+      conf:
+        skipMTLS: false
+        port: 5670
+        path: /metrics
+        envoyFilterRegex: "envoy.cluster.*"
+```
+
+Implementation:
+
+Envoy can receive one [filter with regex](https://www.envoyproxy.io/docs/envoy/latest/operations/admin#get--stats?filter=regex)
+_Note:_ filter works on statsd names, not prometheus names, when configuring filter you need to take that into account. 
+You can only provide one filter, so `envoyFilterRegex` cannot be a list.
+
+Then, in prometheus_endpoint_generator.go we can configure route to add this query param, so the `:5670` is still `/metrics`, but it adds the filter under the hood.
+Then, metrics hijacker receives a request with query param and there are two options:
+* It passes it only to Envoy.
+* It passes it to Envoy and all the apps specified in aggregate.
+I think the first option is safer.
```

Comment:
Agreed I think the 2nd option is a different feature and we'd want to have a filter per "aggregate"